### PR TITLE
Adding Usage Scenarios

### LINKED ISSUE #17: add use cases
In [his review](https://mailarchive.ietf.org/arch/msg/cose/PzdSsG8GP_S2pguJpPOHyIxdHZY/), Orie commented that: _"Some use cases for the 2 modes might improve the document."_

E.g., C2PA, SCITT, PDF timestamping.

### PR DESCRIPTION
**Pull request by henkbirkholz**
Fix #17

### CODE REVIEW

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 1]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
 RFC 3161 {{-TSA}} provides a method to timestamp a message digest to prove that it was created before a given time.
 
 This document defines a new CBOR Object Signing and Encryption (COSE) {{-COSE}} header parameter that carries the TimestampToken (TST) output of RFC 3161, thus allowing existing and widely deployed trust infrastructure to be used with COSE structures used for signing (COSE_Sign and COSE_Sign1).
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:61]: I'd split this into two and add the respective bits to the last line of the use case sections.

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 2]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
 RFC 3161 {{-TSA}} provides a method to timestamp a message digest to prove that it was created before a given time.
 
 This document defines a new CBOR Object Signing and Encryption (COSE) {{-COSE}} header parameter that carries the TimestampToken (TST) output of RFC 3161, thus allowing existing and widely deployed trust infrastructure to be used with COSE structures used for signing (COSE_Sign and COSE_Sign1).
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:60]: Suggest adding a NL here, they are two separate paragraphs.

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 3]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
+
+## Usage Scenario 1: A TST Included in Signed PDF Documents {#use-one}
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:63]: `ETITLETOOLONG` :-D

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 4]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+
+In support of legal assurances, the quality of a signed PDF document can be improved by including a trustworthy signed timestamp.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:65]: _"quality"_ doesn't sound like the right word

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 5]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+In essence, a PDF signer wants to strengthen the assertion that a PDF was not signed before a certain point in time ("the signature cannot be older than").
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:66]: I find this a bit hard to digest.

You want to prove that _"a certain document existed at a certain date"_ in a way that can't be disputed by a relying party.

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 6]
```diff
@@ -54,14 +57,35 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+To achieve this goal, the PDF signer acquires a TST from a TSA, includes it in the to-be-signed PDF that becomes the COSE payload and then signs it.
+Using only its local clock instead, a PDF signer would be able to tamper with its clock and thereby falsify the recentness of the signing procedure.
+
+This usage scenario motivates the "Timestamp then COSE" mode below.
+
+## Usage Scenario 2: Registering a Signed Statement at a Transparency Service {#use-two}
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:72]: ditto

File: draft-birkholz-cose-tsa-tst-header-parameter.md (hunk 2 above)
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:60]:
```suggestion
To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.

```

File: draft-birkholz-cose-tsa-tst-header-parameter.md (hunk 3 above)
Comment (thomas-fossati) [reply draft-birkholz-cose-tsa-tst-header-parameter.md:64]: Sarcasm aside, I will propose text.

Comment (henkbirkholz) [reply draft-birkholz-cose-tsa-tst-header-parameter.md:64]: :heart:

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 7]
```diff
@@ -54,14 +57,36 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
 RFC 3161 {{-TSA}} provides a method to timestamp a message digest to prove that it was created before a given time.
 
 This document defines a new CBOR Object Signing and Encryption (COSE) {{-COSE}} header parameter that carries the TimestampToken (TST) output of RFC 3161, thus allowing existing and widely deployed trust infrastructure to be used with COSE structures used for signing (COSE_Sign and COSE_Sign1).
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
+
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
+
+## Usage Scenario 1: A TST Included in Signed PDF Documents {#use-one}
+
+In support of legal assurances, the quality of a signed PDF document can be improved by including a trustworthy signed timestamp.
```
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:66]:
```suggestion
In support of legal assurances, a signed PDF document can be augmented with a trustworthy signed timestamp.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 8]
```diff
@@ -54,14 +57,36 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
+
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
+
+## Usage Scenario 1: A TST Included in Signed PDF Documents {#use-one}
+
+In support of legal assurances, a signed PDF document can be augmented with a trustworthy timestamp.
+In essence, a PDF signer wants to strengthen the assertion that a PDF was not signed before a certain point in time ("the signature cannot be older than").
+To achieve this goal, the PDF signer acquires a TST from a TSA, includes it in the to-be-signed PDF that becomes the COSE payload and then signs it.
+Using only its local clock instead, a PDF signer would be able to tamper with its clock and thereby falsify the recentness of the signing procedure.
```
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:69]:
```suggestion
Using only its local clock instead, a document signer would be able to tamper with its clock and thereby falsify the recentness of the signing procedure.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 9]
```diff
@@ -54,14 +57,36 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
 RFC 3161 {{-TSA}} provides a method to timestamp a message digest to prove that it was created before a given time.
 
 This document defines a new CBOR Object Signing and Encryption (COSE) {{-COSE}} header parameter that carries the TimestampToken (TST) output of RFC 3161, thus allowing existing and widely deployed trust infrastructure to be used with COSE structures used for signing (COSE_Sign and COSE_Sign1).
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
+
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
+
+## Usage Scenario 1: A TST Included in Signed PDF Documents {#use-one}
+
+In support of legal assurances, a signed PDF document can be augmented with a trustworthy timestamp.
```
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:66]:
```suggestion
In support of legal assurances, a signed document can be augmented with a trustworthy timestamp.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 10]
```diff
@@ -54,14 +57,36 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
 RFC 3161 {{-TSA}} provides a method to timestamp a message digest to prove that it was created before a given time.
 
 This document defines a new CBOR Object Signing and Encryption (COSE) {{-COSE}} header parameter that carries the TimestampToken (TST) output of RFC 3161, thus allowing existing and widely deployed trust infrastructure to be used with COSE structures used for signing (COSE_Sign and COSE_Sign1).
+To motivate the two different modes of use that are specified in this documents, two usage scenarios are illustrated in {{use-one}} and {{use-two}} below.
+
+{{sec-timestamp-then-cose}} and {{sec-cose-then-timestamp}} then define the corresponding modes of use.
+
+## Usage Scenario 1: A TST Included in Signed PDF Documents {#use-one}
```
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:64]:
```suggestion
## Usage Scenario 1: A TST Included in Signed Documents {#use-one}
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 11]
```diff
@@ -54,14 +57,36 @@ This document defines a CBOR Signing And Encrypted (COSE) header parameter for i
...
+
+## Usage Scenario 2: Registering a Signed Statement at a Transparency Service {#use-two}
+
+Transparency services as described in {{-SCITT}} notarize COSE-signed statements by registering them in verifiable data structures that constitute append-only logs.
+Once a signed statement is registered, it cannot be changed.
+In some applications, the registration policy of a transparency service could enforce adding a timestamp to the unprotected header of the COSE-signed statement in order to strengthen the assurance at which point of time the registration occurred ("the registration cannot have happened before").
+To achieve this goal, the transparency service acquires a TST from a TSA, includes it in the unprotected header of the COSE-signed statement and then registers it.
+Using only its local clock instead, a transparency service would be able to tamper with its clock an thereby falsify the recentness of the registration procedure.
+
+This usage scenario motivates the "COSE then Timstamp" mode below.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:81]:
```suggestion
This usage scenario motivates the "COSE then Timestamp" mode below.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 12]
```diff
@@ -71,11 +71,11 @@ This usage scenario motivates the "Timestamp then COSE" mode defined in {{sec-ti
 
 A second use case is the notarization of a signed document by registering it at a Transparency Service.
 This is common for accountability and auditability of issued documents.
-Once a document is registered at a Transparency Service's append-only log, it cannot be changed.
-In certain cases, the registration policy of a Transparency Service may require adding a trustworthy timestamp to the document at the time of registration.
-This is done to enhance confidence in the timing of the registration, ensuring that the registration could not have occurred before a certain point in time.
-To achieve this, the Transparency Service acquires a TST from a TSA, bundles it alongside the signed document, and then registers it.
-A relying party that wants to ascertain the time of registration of a given document does not have to trust the Transparency Service's clock, which may have been maliciously altered or can simply be inaccurate.
+Once a document is registered at a Transparency Service's append-only log, it's log entry cannot be changed.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:74]:
```suggestion
Once a document is registered at a Transparency Service's append-only log, its log entry cannot be changed.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 13]
```diff
@@ -71,11 +71,11 @@ This usage scenario motivates the "Timestamp then COSE" mode defined in {{sec-ti
...
+In certain cases, the registration policy of a Transparency Service may add a trustworthy timestamp to the document to show that the signature could also be validated at the time of registration.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:75]: removing the (redundant) bit that is already explained in L76

```suggestion
In certain cases, the registration policy of a Transparency Service may add a trustworthy timestamp to the signed document.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 14]
```diff
@@ -45,23 +45,47 @@ entity:
...
+Later on, a relying party consuming the signed bundle can be certain that the document existed _at least_ at the time specified by the TSA.
+The relying party does not have to trust the signer's clock, which may have been maliciously altered or simply inaccurate.
+
+This usage scenario motivates the "Timestamp then COSE" mode defined in {{sec-timestamp-then-cose}}.
+
+A second use case is the notarization of a signed document by registering it at a Transparency Service.
+This is common for accountability and auditability of issued documents.
+Once a document is registered at a Transparency Service's append-only log, its log entry cannot be changed.
+In certain cases, the registration policy of a Transparency Service may add a trustworthy timestamp to the signed document.
+This is done to enhance confidence in the authenticity of the document at the time of registration, enshrining that a signature was successfully validated.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:76]:
```suggestion
This is done to seal the signature to a specific point in time.
```

Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:76]:
```suggestion
This is done to seal the association of the signature with a specific point in time.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 15]
```diff
@@ -45,23 +45,47 @@ entity:
...
+The relying party does not have to trust the signer's clock, which may have been maliciously altered or simply inaccurate.
+
+This usage scenario motivates the "Timestamp then COSE" mode defined in {{sec-timestamp-then-cose}}.
+
+A second use case is the notarization of a signed document by registering it at a Transparency Service.
+This is common for accountability and auditability of issued documents.
+Once a document is registered at a Transparency Service's append-only log, its log entry cannot be changed.
+In certain cases, the registration policy of a Transparency Service may add a trustworthy timestamp to the signed document.
+This is done to lock the signature to a specific point in time.
+To achieve this, the Transparency Service acquires a TST from a TSA when its registration policy is met, bundles it alongside the signed document, and then registers it.
```
Comment (henkbirkholz) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:77]:
```suggestion
To achieve this, the Transparency Service acquires a TST from a TSA, bundles it alongside the signed document, and then registers it.
```

File: draft-birkholz-cose-tsa-tst-header-parameter.md [hunk 16]
```diff
@@ -45,23 +45,47 @@ entity:
...
+
+This usage scenario motivates the "Timestamp then COSE" mode defined in {{sec-timestamp-then-cose}}.
+
+A second use case is the notarization of a signed document by registering it at a Transparency Service.
+This is common for accountability and auditability of issued documents.
+Once a document is registered at a Transparency Service's append-only log, its log entry cannot be changed.
+In certain cases, the registration policy of a Transparency Service may add a trustworthy timestamp to the signed document.
+This is done to lock the signature to a specific point in time.
+To achieve this, the Transparency Service acquires a TST from a TSA, bundles it alongside the signed document, and then registers it.
+A relying party that wants to ascertain the authenticity of a document at a certain time does not have to trust the Transparency Service's clock, which may have been maliciously altered or can simply be inaccurate.
```
Comment (thomas-fossati) [inline draft-birkholz-cose-tsa-tst-header-parameter.md:78]:
```suggestion
A relying party that wants to ascertain the authenticity of a document at a subsequent point in time after the signing key has been compromised, can do so by making sure that no revocation information had been made public before the time asserted in the TST.
```

**Review by thomas-fossati — APPROVED** [review]
given enough guidance you can pass as an artist ;)

**Pull request merged**